src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,12 +8,15 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+	public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+	public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses is in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE";
+	public static final String ADDRESS_VALIDATION_REGEX = "(\\d+),\\s(.+),\\s(#\\d+-\\d+),\\s(\\d+)";
 
-    public final String value;
-    private boolean isPrivate;
+	private boolean isPrivate;
+	private Block block;
+	private Street street;
+	private Unit unit ;
+	private PostalCode postalcode;
 
```

Comment:
consider adding actual getters and setters instead hadnling them by modifying the toString().